packages/entities/src/lib/many.query.ts

```diff
@@ -95,3 +97,45 @@ export function selectMany<S extends EntitiesState<Ref>, Ref>(
     distinctUntilArrayItemChanged()
   );
 }
+
+export function selectManyByPredicate<
+  S extends EntitiesState<Ref>,
+  R extends getEntityType<S, Ref>[],
+  K extends keyof getEntityType<S, Ref>,
+  Ref extends EntitiesRef = DefaultEntitiesRef
+>(
+  predicate: ItemPredicate<getEntityType<S, Ref>>,
+  options?: {
+    pluck?: K | ((entity: getEntityType<S, Ref>) => R);
+  } & BaseEntityOptions<Ref>
+): OperatorFunction<S, getEntityType<S, Ref>[]> {
+  const { ref: { entitiesKey } = defaultEntitiesRef, pluck } = options || {};
+
+  return pipe(
+    select<S, getEntityType<S, Ref>[]>((state) => state[entitiesKey]),
+    map((entities) => {
+      const filtered: typeof entities = [];
+
+      Object.values(entities).forEach((entity, index) => {
```

Comment:
We have `findIdsByPredicate` function

Reply:
That's true. We conditionally want to add pluck though. I don't want to loop over the filtered ids again.